src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -37,7 +38,7 @@ object ProtocPlugin extends AutoPlugin with Compat {
         SettingKey[Seq[File]]("protoc-sources", "Directories to look for source files")
       val targets = SettingKey[Seq[Target]]("protoc-targets", "List of targets to generate")
 
-      val runProtoc = SettingKey[Seq[String] => Int](
+      val runProtoc = TaskKey[Seq[String] => Int](
```

Comment:
a settingkey can be assigned to a taskkey from what I understand, so this should not be breaking?

Reply:
What's the motivation for this change? Settings are preferred, since TaskKey introduce a limitation. Other `SettingKey`s can not depend on `TaskKey`s, so if a user or another plugin needs to set some other setting to a value that depends on `runProtoc` they are stuck.  See https://github.com/sbt/sbt-assembly/issues/196

Reply:
Indeed, what I meant in my comment was this change is not breaking for clients redefining the key (1), but it clearly is for clients referencing it from a `TaskKey` as you explain (2). I has assumed the function had been captured/isolated in a key for the purpose of (1) (for example, we do redefine it in order to run some `rm` on some of the `protoc` output to skip unnecessary), but I guess (2) can also exist (although I can't really think of a use case?).

### Rationale

Switching to `TaskKey` was required in this implementation to be able to capture/use the [logger of another task](https://github.com/thesamet/sbt-protoc/pull/121/commits/55aaa055131d72309c548b5b0991af0bd5c20878#diff-608890ca313eef1990a1b4bb7bb3d467R146). Without this, only [`sLog`](https://github.com/sbt/sbt/blob/v1.3.3/main/src/main/scala/sbt/Keys.scala#L60) is available, which does help to replay the logs of a certain `protocGenerate` run via `last protocGenerate`.

### Alternatives

Sticking to `SettingKey` would be possible, but with a different signature: 
```suggestion
      val runProtoc = SettingKey[(Seq[String], Logger) => Int](
```
This way, the caller could inject a context-specific logger. But this is even more breaking as it impacts (1).

Another option I considered was to leave `runProtoc` unchanged (both the key signature and implementation via the [current overload of `com.github.os72.protocjar.Protoc.runProtoc`](https://github.com/os72/protoc-jar/blob/de935a1/src/main/java/com/github/os72/protocjar/Protoc.java#L55)), but move the stream capturing introduced in the next commit to the call site (so within `sourceGeneratorTask`, instead of `runProtocTask`), to map both stdout & stderr to specific log levels during the run of `runProtoc`. Unfortunately, that means that we can't distinguish between `protoc-jar` output (which I find very verbose) vs `protoc` standard output. It felt strange to use the stream capturing hack even though there is [a dedicated overload](https://github.com/os72/protoc-jar/blob/de935a1/src/main/java/com/github/os72/protocjar/Protoc.java#L58) to capture `protoc` logs properly.

Reply:
I prefer the last option you provided. Maybe it's possible to add a new overload as suggested below to protoc-jar.

Reply:
Could you clarify what you had in mind? Because if we go for the second/last alternative, we can't make use of any `protoc-jar` overload (whether the existing one to specifically capture `protoc` stdout/stderr, or a potential new one with an extra output stream for `protoc-jar` own debug logs).

Reply:
On second thought, maybe it would not be too bad to turn `runProtoc` into a task key like you originally suggested.